driver/src/main/java/org/neo4j/driver/internal/util/MetadataExtractor.java

```diff
@@ -99,12 +102,26 @@ public long extractResultAvailableAfter( Map<String,Value> metadata )
     public ResultSummary extractSummary( Statement statement, Connection connection, long resultAvailableAfter, Map<String,Value> metadata )
     {
         ServerInfo serverInfo = new InternalServerInfo( connection.serverAddress(), connection.serverVersion() );
-        return new InternalResultSummary( statement, serverInfo, extractStatementType( metadata ),
-                extractCounters( metadata ), extractPlan( metadata ), extractProfiledPlan( metadata ),
-                extractNotifications( metadata ), resultAvailableAfter, extractResultConsumedAfter( metadata, resultConsumedAfterMetadataKey ) );
+        DatabaseInfo dbInfo = extractDatabaseInfo( metadata );
+        return new InternalResultSummary( statement, serverInfo, dbInfo, extractStatementType( metadata ), extractCounters( metadata ), extractPlan( metadata ),
+                extractProfiledPlan( metadata ), extractNotifications( metadata ), resultAvailableAfter,
+                extractResultConsumedAfter( metadata, resultConsumedAfterMetadataKey ) );
     }
 
-    public Bookmarks extractBookmarks( Map<String,Value> metadata )
+    public static DatabaseInfo extractDatabaseInfo( Map<String,Value> metadata )
+    {
+        Value dbValue = metadata.get( "db" );
+        if ( dbValue == null || dbValue.isNull() || !dbValue.hasType( TYPE_SYSTEM.STRING() ) )
```

Comment:
Shouldn't we throw `ProtocolException` if `!dbValue.hasType( TYPE_SYSTEM.STRING() )` holds?